### DISCUSSION TITLE
[Unreal]请问引擎类继承，有什么方法能从生产的BP再找回对应的JS类对象？

### DISCUSSION THREAD

**Discussion by watsonsong** [Tencent/puerts#1438] [Q&A] [answered]
RT，有时候从BP的引用变量等地方可能拿到一个引擎类继承的对象，也就是生成在Content/Blueprints/TypeScript下面的BP以及他们的子类。
有什么方法能从这些引用中，找回对应的JS类对象么？
有一个blueprint.tojs的方法，但这个是转换类型UClass的。希望有一个能转化实例的方法。

虽然可以把要访问的方法或者属性都暴露成蓝图变量，但如果纯JS访问是否会更高效一些在已知蓝图一定是一个JS对象的情况下。

**Comment by chexiongsheng** [comment]
直接as一下

**Comment by watsonsong** [comment]
直接as似乎在报错，例如
```export default class MyClass extends UE.Object {}```
生成的ue_bp.d.ts中的类型是
```class MyClass _C extends UE.Object```

直接as的话ts的类型检查会报错，还是说其实JS层面可以直接转换使用？

**Reply by chexiongsheng** [reply, marked as answer]
as any as MyClass

**Reply by watsonsong** [reply]
多谢，确实这样就行了~
